src/compiler/checker.ts

```diff
@@ -3874,28 +3910,25 @@ namespace ts {
             return links.resolvedType;
         }
 
-        function addTypeToSortedSet(sortedSet: Type[], type: Type) {
-            if (type.flags & TypeFlags.Union) {
-                addTypesToSortedSet(sortedSet, (<UnionType>type).types);
+        function addTypeToSet(typeSet: Type[], type: Type, typeKind: TypeFlags) {

```

Comment:
I would call it typeSetKind, rather than typeKind, because typeKind sounds like it means `type.flags`.